packages/terra-alert/src/Alert.jsx

```diff
@@ -0,0 +1,273 @@
+import IconRequired from 'terra-icon/lib/icon/IconRequired';
+import IconDiamond from 'terra-icon/lib/icon/IconDiamond';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
+import './Alert.scss';
+
+const AlertTypes = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  ADVISORY: 'advisory',
+  INFO: 'info',
+  SUCCESS: 'success',
+  GAP_CHECKING: 'gap_checking',
+  OUTSIDE_RECORDS: 'outside_records',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the alert.
+   */
+  alertAction: PropTypes.element,
+  /**
+   * Child Nodes providing the message content for the alert. Can contain text and HTML.
+   */
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
+  /**
+   * The icon to be used for an alert of type custom.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * The status bar color to be used for an alert of type custom.
+   */
+  customStatusColor: PropTypes.string,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause the Dismiss button to be included on the alert.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The title for the alert which will be bolded.
+   */
+  title: PropTypes.string,
+  /**
+   * The type of alert to be rendered. One of Alert.Types.ALERT, Alert.Types.ERROR, Alert.Types.WARNING, Alert.Types.ADVISORY,
+   * Alert.Types.INFO, Alert.Types.SUCCESS, Alert.Types.GAP_CHECKING, Alert.Types.OUTSIDE_RECORDS, Alert.Types.CUSTOM.
+   * Use the Types attribute of the Alert component for access to these type strings.
+   */
+  type: PropTypes.oneOf([
+    AlertTypes.ALERT,
+    AlertTypes.ERROR,
+    AlertTypes.WARNING,
+    AlertTypes.ADVISORY,
+    AlertTypes.INFO,
+    AlertTypes.SUCCESS,
+    AlertTypes.GAP_CHECKING,
+    AlertTypes.OUTSIDE_RECORDS,
+    AlertTypes.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  alertAction: null,
+  children: '',
+  customIcon: null,
+  customStatusColor: '',
+  onDismiss: null,
+  title: '',
+  type: AlertTypes.ALERT,
+};
+
+const getAlertClassName = (type) => {
+  switch (type) {
+    case AlertTypes.ALERT:
+      return 'terra-Alert--alert';
+    case AlertTypes.ERROR:
+      return 'terra-Alert--error';
+    case AlertTypes.WARNING:
+      return 'terra-Alert--warning';
+    case AlertTypes.ADVISORY:
+      return 'terra-Alert--advisory';
+    case AlertTypes.INFO:
+      return 'terra-Alert--info';
+    case AlertTypes.SUCCESS:
+      return 'terra-Alert--success';
+    case AlertTypes.GAP_CHECKING:
+      return 'terra-Alert--gap-checking';
+    case AlertTypes.OUTSIDE_RECORDS:
+      return 'terra-Alert--outside-records';
+    case AlertTypes.CUSTOM:
+      return 'terra-Alert--custom';
+    default:
+      return '';
+  }
+};
+
+const getAlertIcon = (type, customIcon) => {
+  switch (type) {
+    case AlertTypes.ALERT:
```

Comment:
The eight standard Alert types are predefined in the UX design and would be mutually exclusive.  If the consuming application has a need for an Alert that does not fit the standard types, there is also a custom alert type where the application can specify the status bar color and the icon to be used for the alert.  And, just like any of the alert types, the application can specify the title for the custom alert, but there is no default title for a custom alert.  Does that answer your question?